### PR TITLE
LibWeb+WebWorker: Move worker execution into a WebWorker process 

### PR DESCRIPTION
**Pull request by ADKaster**
We now create a WorkerAgent for the parent context, which is currently
only a Window. Note that Workers can have Workers per the spec.

The WorkerAgent spawns a WebWorker process to hold the actual
script execution of the Worker. This is modeled with the
DedicatedWorkerHost object in the WebWorker process.
A start_dedicated_worker IPC method in the WebWorker IPC creates the
WorkerHost object. Future different worker types may use different IPC
messages to create their WorkerHost instance.

This implementation cannot yet postMessage between the parent and the
child processes.

But, it can load and execute classic scripts at least :D

<details>

<summary> Logs logs logs </summary>

```
ResourceLoader: Finished load of: "file:///home/andrew/serenity/Base/res/html/misc/worker_parent.html", Duration: 13ms
(js log) "Page Loaded. Sending message"
(js log) "Parent Keys: " "[]"
FontDatabase::load_all_fonts_from_uri('file:///home/andrew/.local/share/fonts'): open: No such file or directory (errno=2)
Fetching ur skript
ResourceLoader: Starting load of: "file:///home/andrew/serenity/Base/res/html/misc/worker.js"
ResourceLoader: Finished load of: "file:///home/andrew/serenity/Base/res/html/misc/worker.js", Duration: 0ms
(js log) "In Worker - Loaded" WorkerGlobalScope{ "onmessage": [Function] onmessage, "self": <already printed Object 0x000055d037b90040>, "location": WorkerLocation{}, "navigator": WorkerNavigator{ "appCodeName": "Mozilla", "appName": "Netscape", "appVersion": "5.0 (Linux; x86_64) Ladybird/1.0", "platform": "Linux x86_64", "product": "Gecko", "productSub": "20030107", "userAgent": "Mozilla/5.0 (Linux; x86_64) Ladybird/1.0", "vendor": "Google Inc.", "vendorSub": "", "onLine": true, "language": "en-US", "languages": [ "en-US" ], "hardwareConcurrency": 1 }, "onerror": null, "onlanguagechange": null, "onoffline": null, "ononline": null, "onrejectionhandled": null, "onunhandledrejection": null, "origin": "null", "isSecureContext": false, "crossOriginIsolated": false, "importScripts": [NativeFunction] , "clearTimeout": [NativeFunction] , "btoa": [NativeFunction] , "setTimeout": [NativeFunction] , "setInterval": [NativeFunction] , "atob": [NativeFunction] , "structuredClone": [NativeFunction] , "clearInterval": [NativeFunction] , "fetch": [NativeFunction] , "queueMicrotask": [NativeFunction] , "removeEventListener": [NativeFunction] , "addEventListener": [NativeFunction] , "dispatchEvent": [NativeFunction]  }
"In Worker - Loaded" WorkerGlobalScope{ "onmessage": [Function] onmessage, "self": <already printed Object 0x000055d037b90040>, "location": WorkerLocation{}, "navigator": WorkerNavigator{ "appCodeName": "Mozilla", "appName": "Netscape", "appVersion": "5.0 (Linux; x86_64) Ladybird/1.0", "platform": "Linux x86_64", "product": "Gecko", "productSub": "20030107", "userAgent": "Mozilla/5.0 (Linux; x86_64) Ladybird/1.0", "vendor": "Google Inc.", "vendorSub": "", "onLine": true, "language": "en-US", "languages": [ "en-US" ], "hardwareConcurrency": 1 }, "onerror": null, "onlanguagechange": null, "onoffline": null, "ononline": null, "onrejectionhandled": null, "onunhandledrejection": null, "origin": "null", "isSecureContext": false, "crossOriginIsolated": false, "importScripts": [NativeFunction] , "clearTimeout": [NativeFunction] , "btoa": [NativeFunction] , "setTimeout": [NativeFunction] , "setInterval": [NativeFunction] , "atob": [NativeFunction] , "structuredClone": [NativeFunction] , "clearInterval": [NativeFunction] , "fetch": [NativeFunction] , "queueMicrotask": [NativeFunction] , "removeEventListener": [NativeFunction] , "addEventListener": [NativeFunction] , "dispatchEvent": [NativeFunction]  }
(js log) "Keys: " "["onmessage"]"
"Keys: " "["onmessage"]"
Unhandled JavaScript exception: [ReferenceError] 'postMessage' is not defined
    at file:///home/andrew/serenity/Base/res/html/misc/worker.js:8:12
```

</details>

### CODE REVIEW

File: Userland/Libraries/LibWebView/WebContentClient.cpp [hunk 1]
```diff
@@ -395,6 +395,16 @@ void WebContentClient::did_request_file(DeprecatedString const& path, i32 reques
...
+
+    // FIXME: Hax to make local workers work, remove this!
+    if (!m_view) {
+        auto file = Core::File::open(path, Core::File::OpenMode::Read);
+
+        if (file.is_error())
+            async_handle_file_return(file.error().code(), {}, request_id);
+        else
+            async_handle_file_return(0, IPC::File(*file.value()), request_id);
+    }
```
Comment (ADKaster) [inline Userland/Libraries/LibWebView/WebContentClient.cpp:407]: This is particularly egregious, but the alternative is to invent that "WorkerView" thing that feels like overkill.

File: Userland/Libraries/LibWeb/HTML/WorkerAgent.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+WorkerAgent::WorkerAgent(AK::URL url, WorkerOptions const& options)
+    : m_worker_options(options)
+    , m_url(move(url))
+{
+#ifndef AK_OS_SERENITY
+    // FIXME: Add factory function
+    auto paths = MUST(get_paths_for_helper_process("WebContent"sv));
+    m_worker_ipc = MUST(launch_web_content_process(paths));
+#else
+    m_worker_ipc = MUST(WebView::WebContentClient::try_create());
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/HTML/WorkerAgent.cpp:104]: From a quick look at the build failure - LibWeb cannot depend on LibWebView

Comment (ADKaster) [reply Userland/Libraries/LibWeb/HTML/WorkerAgent.cpp:104]: Bleh. I was hoping we could hack around that. Looks like this'll need a custom WebContentClient then

Comment (trflynn89) [reply Userland/Libraries/LibWeb/HTML/WorkerAgent.cpp:104]: Could this instead be an IPC that the chrome layer receives to create a worker process?

**Comment by ADKaster** [comment]
I'll do the WebWorker process idea. At least we know that in theory this works, even if the WhatWG folks think that we'll need to put DedicatedWorkers in the same process as their owner in the future.

**Comment by ADKaster** [comment]
IDK what github is smoking but the state of the PR doesn't match my branch

Branch in my fork
image

PR
image

**Pull request merged**